📝 Update docs: replace Atom references with Pulsar [ci skip] (#1607)
pulsar-edit/pulsar: CONTRIBUTING.md

Explain the problem and include additional details to help maintainers reproduce the problem:

* **Use a clear and descriptive title** for the issue to identify the problem.
* **Describe the exact steps which reproduce the problem** in as many details as possible. For example, start by explaining how you started Pulsar, e.g. which command exactly you used in the terminal, or how you started Pulsar otherwise. When listing steps, **don't just say what you did, but explain how you did it**. For example, if you moved the cursor to the end of a line, explain if you used the mouse, or a keyboard shortcut or a Pulsar command, and if so which one?
* **Provide specific examples to demonstrate the steps**. Include links to files or GitHub projects, or copy/pasteable snippets, which you use in those examples. If you're providing snippets in the issue, use [Markdown code blocks](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github).
* **Describe the behavior you observed after following the steps** and point out what exactly is the problem with that behavior.
* **Explain which behavior you expected to see instead and why.**

> * **Include screenshots and animated GIFs** which show you following the described steps and clearly demonstrate the problem. If you use the keyboard while following the steps, **record the GIF with the [Keybinding Resolver](https://github.com/pulsar-edit/pulsar/tree/master/packages/keybinding-resolver) shown**. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) on Linux.

Reviewer: ```suggestion
* **Include screenshots and/or screencasts** which show you following the described steps and clearly demonstrate the problem. If you use the keyboard while following the steps, **record the screencast with the [Keybinding Resolver](https://github.com/pulsar-edit/pulsar/tree/master/packages/keybinding-resolver) shown**.  The exact program used to record one's screen will vary from platform to platform. If you like, you can use `ffmpeg` to convert your screencast to MP4 for broadest compatibility: e.g., `ffmpeg -i screencast.mov screencast.mp4`.
```

These days screencasts are probably easier to embed than animated GIFs are to make!

> * **If you're reporting that Pulsar crashed**, include a crash report with a stack trace from the operating system. On macOS, the crash report will be available in `Console.app` under "Diagnostic and usage information" > "User diagnostic reports". Include the crash report in the issue in a [code block](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github), a [file attachment](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files), or put it in a [gist](https://gist.github.com/discover) and provide link to that gist.

Reviewer: ```suggestion
* **If you're reporting that Pulsar crashed**, include a crash report with a stack trace from the operating system. On macOS, the crash report will be available in `Console.app` under "Diagnostic and usage information" > "User diagnostic reports". On Windows and Linux, you should be able to find a new crashdump in the `crashdumps` folder within `ATOM_HOME` — e.g., `~/.pulsar/crashdumps` on Linux. Include the crash report in the issue in a [code block](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github), a [file attachment](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files), or put it in a [gist](https://gist.github.com/discover) and provide link to that gist.
```

* **If the problem is related to performance or memory**, include a [CPU profile capture](https://docs.pulsar-edit.dev/troubleshooting-pulsar/diagnosing-runtime-performance/) with your report.

> * **If Chrome's developer tools pane is shown without you triggering it**, that normally means that you have a syntax error in one of your themes or in your `styles.less`. Try running in [Safe Mode](https://docs.pulsar-edit.dev/troubleshooting-pulsar/using-safe-mode/) and using a different theme or comment out the contents of your `styles.less` to see if that fixes the problem.

Reviewer: ```suggestion
* **If Chrome's developer tools pane is shown without you triggering it**, that typically means that an installed package is throwing an error, or else that there is a syntax error in your `styles.less`. In the former case, running in [Safe Mode](https://docs.pulsar-edit.dev/troubleshooting-pulsar/using-safe-mode/) will disable non-core packages so you can compare the behavior; in the latter case, you can try using a different theme or commenting out the contents of your `styles.less` to see if that fixes the problem.
```

* **Include screenshots and animated GIFs** which help you demonstrate the steps or point out the part of Pulsar which the suggestion is related to. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) on Linux.
* **Explain why this enhancement would be useful** to most Pulsar users and isn't something that can or should be implemented as a [community package](#pulsar-and-packages).
* **List some other text editors or applications where this enhancement exists.**
* **Specify which version of Pulsar you're using.** You can get the exact version by running `pulsar -v` in your terminal, or by starting Pulsar and running the `Application: About` command from the [Command Palette](https://github.com/pulsar-edit/pulsar/tree/master/packages/command-palette).
* **Specify the name and version of the OS you're using.**

### Your First Code Contribution

Unsure where to begin contributing to Pulsar? You can start by looking through these `beginner` and `help-wanted` issues:

* [Beginner issues][beginner] - issues which should only require a few lines of code, and a test or two.

> * [Help wanted issues][help-wanted] - issues which should be a bit more involved than `beginner` issues.

Reviewer: ```suggestion
* [Help wanted issues][help-wanted] - issues which should be a bit more involved than `beginner` issues. These could be issues that require subject matter expertise that we don't have. They could also be issues that aren't on our roadmap — for which we'd accept a well-written pull request, but which otherwise don't have a high chance of being addressed by the core maintainers in the near future.
```

* Reference issues and pull requests liberally after the first line

> * When only changing documentation, include `[ci skip]` in the commit title

Reviewer: ```suggestion
```

I could be wrong, but I think we don't have this set up — and aside from files like these, most documentation will be in its own repo anyway.

* Consider starting the commit message with an applicable emoji:
    * 🎨 `:art:` when improving the format/structure of the code
    * 🐎 `:racehorse:` when improving performance
    * 🚱 `:non-potable_water:` when plugging memory leaks
    * 📝 `:memo:` when writing docs
    * 🐧 `:penguin:` when fixing something on Linux
    * 🍎 `:apple:` when fixing something on macOS
    * 🏁 `:checkered_flag:` when fixing something on Windows
    * 🐛 `:bug:` when fixing a bug
    * 🔥 `:fire:` when removing code or files
    * 💚 `:green_heart:` when fixing the CI build
    * ✅ `:white_check_mark:` when adding tests
    * 🔒 `:lock:` when dealing with security
    * ⬆️ `:arrow_up:` when upgrading dependencies
    * ⬇️ `:arrow_down:` when downgrading dependencies

>     * 👕 `:shirt:` when removing linter warnings

Reviewer: We have no track record of using any of these emoji. I wouldn't begrudge their use, but it doesn't feel like it's worth suggesting to new contributors, since they'd basically be the only ones using these icons!

Author: Ok, will remove.

  ```
* Place requires in the following order:

>     * Built in Node Modules (such as `path`)

Reviewer: ```suggestion
    * Built-in Node Modules (such as `path`)
```

>     * Built in Pulsar and Electron Modules (such as `atom`, `remote`)

Reviewer: ```suggestion
    * Built-in Pulsar and Electron Modules (such as `atom`, `remote`)
```

    * Local Modules (using relative paths)
* Place class properties in the following order:
    * Class methods and properties (methods starting with `static`)
    * Instance methods and properties

> * Avoid platform-dependent code

Reviewer: ```suggestion
* Avoid platform-dependent code
  * If you must resort to a platform-dependent fix, you should still ensure that the affected code path works as expected on each of our three supported platforms — and that we prove it with specs to the greatest possible extent
```

> ### CoffeeScript Styleguide

Reviewer: We should be able to lose this entire section, since we wouldn't accept even well-written CoffeeScript in a PR anymore!

### Documentation Styleguide

* Use [AtomDoc](https://github.com/pulsar-edit/atomdoc).

> * Use [Markdown](https://daringfireball.net/projects/markdown/).

Reviewer 2: ```suggestion
* Use [Markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github).
```

These days we defer to GitHub's handling of Markdown in every instance, so it makes sense to actually link to them.

### Documentation Styleguide

> * Use AtomDoc.

Reviewer: ```suggestion
* Use [AtomDoc](https://github.com/pulsar-edit/atomdoc).
```

This will do as a link target for now — the README has enough information in it to be useful to a new contributor.
